src/UI/class-settings-page.php

```diff
@@ -1294,4 +1328,63 @@ private function get_unobfuscated_value( $current_value, $previous_value ): stri
 
 		return $current_value;
 	}
+
+	/**
+	 * Returns the field label's title. If the option is managed, a badge is
+	 * also included.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @param string $title The field's title.
+	 * @param string $option_id The option's ID.
+	 *
+	 * @return string The resulting content.
+	 */
+	public function set_field_label_contents( string $title, string $option_id ): string {
+		$is_managed = key_exists( $option_id, $this->parsely->managed_options );
+
+		if ( $is_managed ) {
+			$text = $this->managed_options_badge['text'] ?? '';
+			if ( ! is_string( $text ) || '' === $text ) {
+				return $title;
+			}
+
+			$text = esc_attr( $text );
```

Comment:
We should escape as late as possible. Additionally `esc_attr` is not ideal for non-attribute content.

```suggestion
```

Reply:
The context here is that we don't want HTML passed into the badge, and this is why `esc_attr()` was picked. Can switch this to `esc_html` if this is what you think we should do.
I prefer to escape the variable before since it is only for display purposes, but since this is a WP convention I'll adjust to that.

Reply:
Applied your suggestions below using `esc_html()` in https://github.com/Parsely/wp-parsely/pull/1724/commits/d5a58d5378e80a8be9980819d53c51797ceb59ae. Marking the related suggestions as resolved, and keeping this open for context.